lib/reader.ts

```diff
@@ -437,17 +455,18 @@ class ParquetEnvelopeReader {
   }
 
   // Helper function to get the column object for a particular path and row_group
-  getColumn(path, row_group) {
+  getColumn(path: string | ColumnData, row_group: RowGroup | number | null) {
     let column;
-    if (!isNaN(row_group)) {
-      row_group = this.metadata.row_groups[row_group];
+    if (!isNaN(row_group as number)) {
+      row_group = this.metadata!.row_groups[row_group as number];
```

Comment:
Do not change. Side note only.

This should likely be doing [type narrowing](https://www.typescriptlang.org/docs/handbook/2/narrowing.html) instead of a NaN check. That would correctly resolve several of the `as`s here and below.

Then you could refactor this to be a `switch(true)` statement:

```typescript
switch(true) {
  case isNumber(rowGroup):
    // ...
    break;
  case (typeof path === "string" && isRowGroup(rowGroup)):
    // ...
    break;
  default:
    // throw
}
```